torch/ao/quantization/observer.py

```diff
@@ -454,8 +458,8 @@ def extra_repr(self):
     @torch.jit.export
     def reset_min_max_vals(self):
         """Resets the min/max values."""
-        self.min_val = torch.tensor(float("inf"))
-        self.max_val = torch.tensor(float("-inf"))
+        self.min_val.copy_(torch.tensor(float("inf")))
+        self.max_val.copy_(torch.tensor(float("-inf")))
 
```

Comment:
this is where the change to preserve the characteristics (device most importantly) of self.min_val and self.max_val was made since the way those values are assigned in the init statement uses factory_kwargs